### PR TITLE
api: deprecate BuildCache.Parent, add BuildCache.Parents in API >= v1.42

### PR DESCRIPTION
**Pull request by thaJeztah**
First two commits are documenting the fields

- relates to https://github.com/moby/buildkit/commit/d73e62f878746bb7d8f73fdd0d7bd3043c8b5f5f (https://github.com/moby/buildkit/pull/2335)

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
@crazy-max @tonistiigi PTAL

**Review by tianon — APPROVED** [review]

**Comment by thaJeztah** [comment]
Having some discussion on Slack; looks like the replacement is a Parents (plural) field, so we may have to add that one as well

**Comment by thaJeztah** [comment]
> Having some discussion on Slack; looks like the replacement is a Parents (plural) field, so we may have to add that one as well

I added the `Parents` field, and updated the documentation

@tianon @tonistiigi @crazy-max PTAL

**Comment by crazy-max** [comment]
SGTM thanks!

File: docs/api/version-history.md [hunk 1]
```diff
@@ -87,6 +87,10 @@ keywords: "API, Docker, rcli, REST, documentation"
   volume (CNI). This option can only be used if the daemon is a Swarm manager.
   The Volume response on creation now also can contain a `ClusterVolume` field
   with information about the created volume.
+* The `BuildCache.Parent` field, as returned by `GET /system/df` is deprecated
+  and is now omitted. API versions before v1.42 continue to include this field.
+* `GET /system/df` now includes a new `Parents` field, for "build-cache" records,
+  which contains a list of parent ID's for the build-cache record.
```
Comment (tonistiigi) [inline docs/api/version-history.md:93]: I believe it is `IDs`

**Review by tonistiigi — APPROVED** [review]

File: docs/api/version-history.md (hunk 1 above)
Comment (thaJeztah) [reply docs/api/version-history.md:93]: Yes, I think you're right; must've been in "dutch" mode while I wrote this 🇳🇱 😁

Comment (thaJeztah) [reply docs/api/version-history.md]: fixed 👍

**Comment by thaJeztah** [comment]
Failure on Windwos 2022 is unrelated, but let me post this as I've seen this error pop-up a few times (could be a platform issue)

```
=== FAIL: github.com/docker/docker/integration/container TestWaitNonBlocked/wait-nonblocking-exit-0 (1.49s)
    wait_test.go:44: assertion failed: error is not nil: Error response from daemon: hcs::CreateComputeSystem 715ba8aab407c4fc98495e33f58ada3af44a9b2431d9ebd3fd1a60ece43db89d: Element not found.
    --- FAIL: TestWaitNonBlocked/wait-nonblocking-exit-0 (1.49s)
```

**Comment by thaJeztah** [comment]
All green now 👍

**Comment by thaJeztah** [comment]
@dgiebert opened https://github.com/moby/moby/pull/43913

**Pull request merged**